### PR TITLE
Containerized billing script

### LINKED ISSUE #21: Create Dockerfile to containerize billing script
Splitting up #15, and dependent on #20, as the next step before creating a fully automated cronjob to process invoices, we want to create a Docker container containing the bare dependencies needed to run the cloning and billing script.

### PR DESCRIPTION
**Pull request by QuanMPhm**
Closes #21 completely. This PR is blocked on #27, and will containerize the billing script. A github workflow is added to build and publish the billing image. Check the second commit to see the changes which actually implemented the containerization.

### CODE REVIEW

File: tools/clone_nonbillables_and_process.sh [hunk 1]
```diff
@@ -28,9 +28,19 @@ if [ ! -d ./non-billable-projects ]; then
 fi
 
 INVOICE_MONTH=$(date --date="$(date +%Y-%m-01) -1 month" +%Y-%m)
+export S3_ENDPOINT
```
Comment (knikolla) [inline tools/clone_nonbillables_and_process.sh:31]: These exports are unnecessary as these valued will not be assigned in the script but outside of it.

File: tools/clone_nonbillables_and_process.sh [hunk 2]
```diff
@@ -28,9 +28,19 @@ if [ ! -d ./non-billable-projects ]; then
...
+    --upload-to-s3 \
     --invoice-month $INVOICE_MONTH \
     --pi-file ./non-billable-projects/pi.txt \
     --projects-file ./non-billable-projects/projects.txt \
     --timed-projects-file ./non-billable-projects/timed_projects.txt \
-    --old-pi-file old_pi.csv
+    --nonbillable-file "NERC (Nonbillable)" \
+    --output-file "NERC" \
+    --output-folder "PI Invoices" \
+    --BU-invoice-file "NERC BU" \
```
Comment (knikolla) [inline tools/clone_nonbillables_and_process.sh:45]: You are missing the file extensions in all arguments that are a file here.

**Review by knikolla — CHANGES_REQUESTED** [review]

File: tools/clone_nonbillables_and_process.sh (hunk 2 above)
Comment (QuanMPhm) [reply tools/clone_nonbillables_and_process.sh:45]: Since all the files in the container will be ephemeral, why do we need to include extensions?

Comment (knikolla) [reply tools/clone_nonbillables_and_process.sh:45]: You are assuming that this script will only ever be used inside the container.

**Review by naved001 — APPROVED** [review]

**Pull request merged**